src/Loop/Driver.php

```diff
@@ -105,6 +105,8 @@ public function onSignal($signo, callable $callback, $data = null);
      * @param string $watcherId The watcher identifier.
      *
      * @return void
+     *
+     * @throws InvalidWatcherException Thrown if the watcher identifier is invalid or cancelled.
```

Comment:
I do not care too much about this; this just was part of the initial proposal and I was explaining to Andrew as he asked.